### DISCUSSION TITLE
Bars with missed rests aren't marked incomplete?

### DISCUSSION THREAD

**Discussion by npostavs** [Audiveris/audiveris#584] [Q&A] [answered]
Version: 5.3-alpha:ee68a8c

I tried running the [Oboe I of TWV 44:43](https://imslp.org/wiki/Concerto_for_3_Oboes,_3_Violins_and_Continuo,_TWV_44:43_(Telemann,_Georg_Philipp)#IMSLP587938) through audiveris and it works pretty well. Just the eighth rests are not recognized. But what confuses me, is that in the bars with eighth rests have missing beats (since the rests are missed) but they are not highlighted in red as shown in https://audiveris.github.io/audiveris/_pages/ui_foundations/inspection/. The key signature seems to be recognized as COMMON_TIME. Toggling `View | Show score Errors`/<kbd>F9</kbd> makes no difference.

I have seen this red highlighting work with other scores, so I don't understand why all of a sudden it is not happening.

[pdf with just 1st page which shows the problem](https://github.com/Audiveris/audiveris/files/9015775/IMSLP587938-PMLP398454-Oboe_I.II.III-page1.pdf)

Screenshots in physical then logical mode:
![image](https://user-images.githubusercontent.com/287742/176576188-8517ea40-59fa-481f-98f1-fcca291da267.png)
![image](https://user-images.githubusercontent.com/287742/176576122-c8542d3c-4f75-467e-b253-3580d64b9094.png)

**Comment by hbitteur** [comment, marked as answer]
@npostavs 
Reading your post, I understand that there are missing rests in some measures.
Since there seems to be a single voice in this example, this results in some measures shorter than the expected measure duration as indicated by the time signature.

Audiveris signals an _abnormal_ measure, and displays its background in pink, only in 2 cases:
- When the time offset of a chord (head or rest) could not be computed for whatever reason
- When a measure voice ends **beyond** the expected measure end.

Measures can be **shorter** than the duration indicated by time signature. Think of introductory measure at the beginning of a system, cautionary measure at system end, half repeat measures.
In other words, a short measure is not always an _abnormal_ measure.
This is the reason why current Audiveris version is not able to reliably detect **_abnormal_ short** measures.

Perhaps we could flag short measures that are not recognized as pertaining to the "legal" short cases listed above.
We might get false positives, in part because voice algorithm is really tricky.

But after all this pink background is just a means to call user attention on the measure. It does not impact the exported MusicXML data.
And we could still let the end user enable or disable this "short measure warning" option.

**Reply by npostavs** [reply]
> When a measure voice ends **beyond** the expected measure end.

Ah, I see now that the documentation clearly says that, but I somehow got the impression the red colour indicated any rhythm mismatch. I guess it feels simpler from a user point of view, but the legal partial measures do make it a bit trickier as you say. Maybe shorter measures could be in yellow, to indicate that it's more of a warning than an error.

**Reply by hbitteur** [reply]
+1 for the yellow color!
